device/api/umd/device/cluster.h

```diff
@@ -666,18 +729,45 @@ class Cluster : public tt_device {
     static std::unordered_map<tt_xy_pair, tt_xy_pair> create_harvested_coord_translation(
         const tt::ARCH arch, bool identity_map);
     std::unordered_map<tt_xy_pair, tt_xy_pair> get_harvested_coord_translation_map(chip_id_t logical_device_id);
-    virtual std::uint32_t get_num_dram_channels(std::uint32_t device_id);
-    virtual std::uint64_t get_dram_channel_size(std::uint32_t device_id, std::uint32_t channel);
-    virtual std::uint32_t get_num_host_channels(std::uint32_t device_id);
-    virtual std::uint32_t get_host_channel_size(std::uint32_t device_id, std::uint32_t channel);
-    virtual std::uint32_t get_numa_node_for_pcie_device(std::uint32_t device_id);
-    virtual tt_version get_ethernet_fw_version() const;
-
-    TTDevice* get_tt_device(chip_id_t device_id) const;
 
-    const tt_SocDescriptor& get_soc_descriptor(chip_id_t chip_id) const;
-    // TODO: This function should be removed.
-    std::unordered_map<chip_id_t, tt_SocDescriptor> get_virtual_soc_descriptors();
+    // New API. UMD is transitioning to use CoreCoord instead of tt_xy_pair.
+    // This is new set of functions that should be used once the transition for clients (tt-metal, tt-lens) is complete.
+    virtual void configure_tlb(
+        chip_id_t logical_device_id,
+        tt::umd::CoreCoord core,
+        int32_t tlb_index,
+        uint64_t address,
+        uint64_t ordering = TLB_DATA::Posted);
+    virtual void deassert_risc_reset_at_core(
+        const chip_id_t chip,
+        const tt::umd::CoreCoord core,
+        const TensixSoftResetOptions& soft_resets = TENSIX_DEASSERT_SOFT_RESET);
+    virtual void assert_risc_reset_at_core(const chip_id_t chip, const tt::umd::CoreCoord core);
+    virtual void write_to_device(
+        const void* mem_ptr,
+        uint32_t size_in_bytes,
+        chip_id_t chip,
+        tt::umd::CoreCoord core,
+        uint64_t addr,
+        const std::string& tlb_to_use);
+    virtual void read_from_device(
+        void* mem_ptr,
+        chip_id_t chip,
+        tt::umd::CoreCoord core,
+        uint64_t addr,
+        uint32_t size,
+        const std::string& fallback_tlb);
+    std::optional<std::tuple<uint32_t, uint32_t>> get_tlb_data_from_target(
+        const chip_id_t chip, const tt::umd::CoreCoord core);
+    tt::Writer get_static_tlb_writer(const chip_id_t chip, const tt::umd::CoreCoord target);
+    virtual tt::umd::CoreCoord to(
```

Comment:
I don't see the base class so not clear to me - why do we need a virtual fn to do the coordinate conversion?
Also, "to" is pretty terse.

Reply:
I've left the same comment about "to".
Regarding virtual, you can see we have "virtual" all around. We should do a sweep and remove all of it from this Cluster class. I'm waiting on enough API changes going through so we can finally remove the tt_device abstract class, and then we'll probably fix this one as well.
You're right that this should not have `virtual`